### PR TITLE
Partial Blocks for Index Serialization

### PR DESCRIPTION
**Pull request by taniabogatsch**
Before this PR, we always write a block for each buffer with index data. If one such buffer happens to have very few allocated segments, we still write an entire block for these few bytes, e.g., if there is only one Node256 in an ART.

In this PR, we use the `PartialBlockManager` to combine these small (partial) blocks into a single block, reducing checkpoint size.

### CODE REVIEW

File: src/execution/index/fixed_size_buffer.cpp [hunk 1]
```diff
@@ -5,31 +5,65 @@
 
 namespace duckdb {
 
+//===--------------------------------------------------------------------===//
+// PartialBlockForIndex
+//===--------------------------------------------------------------------===//
+
+void PartialBlockForIndex::AddUninitializedRegion(idx_t start, idx_t end) {
+	throw InternalException("no uninitialized regions for PartialBlockForIndex");
```
Comment (Mytherin) [inline src/execution/index/fixed_size_buffer.cpp:13]: We should mark regions as uninitialized as discussed - could you add an index test to the storage initialization verification?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one comment:

File: test/sql/index/art/vacuum/test_art_vacuum_integers.test_slow [hunk 2]
```diff
@@ -5,8 +5,6 @@
 statement ok
 PRAGMA enable_verification
 
-load __TEST_DIR__/test_art_vacuum_integers.db
```
Comment (Mytherin) [inline test/sql/index/art/vacuum/test_art_vacuum_integers.test_slow:8]: Is this moved to a non-storage test on purpose?

File: test/sql/index/art/vacuum/test_art_vacuum_strings.test_slow [hunk 3]
```diff
@@ -5,8 +5,6 @@
 statement ok
 PRAGMA enable_verification
 
-load __TEST_DIR__/test_art_vacuum_strings.db
```
Comment (Mytherin) [inline test/sql/index/art/vacuum/test_art_vacuum_strings.test_slow:8]: Is this moved to a non-storage test on purpose?

**Review by Mytherin — COMMENTED** [review]
Thanks for the fixes! One more comment

File: test/sql/index/art/vacuum/test_art_vacuum_integers.test_slow (hunk 2 above)
Comment (taniabogatsch) [reply test/sql/index/art/vacuum/test_art_vacuum_integers.test_slow:8]: Or, I can add these two pragmas to the test instead and then add some manual checkpoints/restarts if we want to keep testing that in these tests. What do you think? 😄 
```sql
statement ok
PRAGMA disable_checkpoint_on_shutdown;

statement ok
PRAGMA wal_autocheckpoint='1TB';
```

Comment (Mytherin) [reply test/sql/index/art/vacuum/test_art_vacuum_integers.test_slow:8]: Fair enough :)

**Pull request merged**